blog/deep-dive-temporal-access-control-strategy-aws.md

```diff
@@ -0,0 +1,178 @@
+---
+slug: deep-dive-temporal-access-control-strategy-aws
+title: A Deep Dive into Temporal's Access Control Strategy in AWS
+author: Brandon Sherman
+author_title: Staff Security Engineer
+author_image_url: /img/brandon-sherman.png
+tags:
+  - security
+  - transparency
+  - cloud
+date: 2022-06-06T00:00:00Z
+image: /img/temporal-access-control.jpeg
+---
+
+This blog post gives some insight into Temporal's strategy for securing our cloud environment.
+
+![Photo of dodgy-looking ATM beside a security grate](/img/temporal-access-control.jpeg)
+
+<!-- truncate -->
+
+It also calls attention to an unexpected facet of AWS access policies encountered along the way. Finally, I'll come around to what I discovered—what I thought I wanted isn't what I needed—and describe what I need.
+
+Our long-term goal is to develop an access control mechanism that can be flexible, scalable, and secure. After we reviewed various strategies, a [Biba-style model](https://en.wikipedia.org/wiki/Biba_Model) of segmenting our AWS accounts seems to fit our needs the best. Each account forms a hard<sup>1</sup> "perimeter" and provides an easy way to set, measure, and maintain the maximum "blast radius" of each account. If we assume an account has been breached, what is the maximum extent of damage an attacker can inflict? 
+
+Setting up numerous smaller accounts helps to constrain possible damage, but this approach would also make our management infrastructure a very compelling target. If only one management account has permissions configured to allow those with access to assume roles into all other accounts, and the management account is breached, the blast radius goes from being extremely small to being extremely large. **Clearly, this is not ideal.**
+
+We want to provide strong isolation guarantees to our customers as well as ourselves. An engineer paged at 3:00 AM needs to be comfortable in the knowledge they will not cause large-scale problems when they fix the isolated problem they were paged about. To provide these strong isolation guarantees, we plan to have a small number of trusted core accounts. Think of these as Ring 0: interacting with the infrastructure of these accounts in any way should be a big event.<sup>2</sup> The next ring of accounts, Ring 1, will only listen to commands from Ring 0 and issue commands to Ring 2. Similarly, our core accounts—those Ring 0 and Ring 1 accounts—should not be able to read the data in those outer rings.
+
+It's easy enough to describe this approach, but promises are meaningful only if you follow through with them! The first step for me was to codify this into our AWS service control policies (SCPs). SCPs are extremely powerful tools because they apply to all principals within an AWS account, including the all-powerful root user. Effective SCP layout and implementation is critical to a well-secured AWS environment.
```

Comment:
```suggestion
It's easy enough to describe this approach, but promises are only meaningful if you follow through with them! The first step was to codify this concept into our AWS service control policies (SCPs). SCPs are extremely powerful tools because they apply to all principals within an AWS account, including the all-powerful root user. Effective SCP layout and implementation is critical to a well-secured AWS environment.
```